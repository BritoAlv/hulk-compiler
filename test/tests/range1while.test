type Range(start : number, end : number, offset : number) {
    start = start;
    end = end;
    current = start - offset;
    offset = offset;

    next(): bool => (self.current := self.current + self.offset) < self.end ;
    current(): number => self.current;
}

let interval = new Range(1, 5, 1) in 
{
    print(interval.next());
    print(interval.current());
    print(interval.next());
    print(interval.current());
    print(interval.next());
    print(interval.current());
    print(interval.next());
    print(interval.current());
    print(interval.next());
    print(interval.current());
    print(interval.next());
    print(interval.current());
};



let i = 0 in
while (i < 10) 
    print(i := i + 1);