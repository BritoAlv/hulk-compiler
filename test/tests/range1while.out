There was a parsing error :
Near error on line 28, let:
26 
27 
28 let i = 0 in
29 while (i < 10) 

28 let i = 0 in
28 ^
Trying to recovery from error
